sdk/core/core/inc/az_span.h

```diff
@@ -401,7 +407,12 @@ typedef struct
  *
  */
 #define AZ_PAIR_NULL \
-  (az_pair) { .key = AZ_SPAN_NULL, .value = AZ_SPAN_NULL }
+  (az_pair) \
+  { \
+    .key = { ._internal = { .ptr = NULL, .size = 0 } }, \
+    .value \
+        = {._internal = { .ptr = NULL, .size = 0 } } \
```

Comment:
Why was this change necesssary?

Reply:
There was a warning, and this is the way to avoid it. I'll get the specific warning and post it here.